src/cargo/util/toml/targets.rs

```diff
@@ -1078,31 +1078,39 @@ Cargo doesn't know which to use because multiple target files found at `{}` and
 #[tracing::instrument(skip_all)]
 pub fn normalize_build(
     build: Option<&TomlPackageBuild>,
+    multiple_build_scripts: bool,
     package_root: &Path,
-) -> Option<TomlPackageBuild> {
+) -> CargoResult<Option<TomlPackageBuild>> {
     const BUILD_RS: &str = "build.rs";
     match build {
         None => {
             // If there is a `build.rs` file next to the `Cargo.toml`, assume it is
             // a build script.
             let build_rs = package_root.join(BUILD_RS);
             if build_rs.is_file() {
-                Some(TomlPackageBuild::SingleScript(BUILD_RS.to_owned()))
+                Ok(Some(TomlPackageBuild::SingleScript(BUILD_RS.to_owned())))
             } else {
-                Some(TomlPackageBuild::Auto(false))
+                Ok(Some(TomlPackageBuild::Auto(false)))
             }
         }
         // Explicitly no build script.
-        Some(TomlPackageBuild::Auto(false)) => build.cloned(),
+        Some(TomlPackageBuild::Auto(false)) => Ok(build.cloned()),
         Some(TomlPackageBuild::SingleScript(build_file)) => {
             let build_file = paths::normalize_path(Path::new(build_file));
             let build = build_file.into_os_string().into_string().expect(
                 "`build_file` started as a String and `normalize_path` shouldn't have changed that",
             );
-            Some(TomlPackageBuild::SingleScript(build))
+            Ok(Some(TomlPackageBuild::SingleScript(build)))
         }
         Some(TomlPackageBuild::Auto(true)) => {
-            Some(TomlPackageBuild::SingleScript(BUILD_RS.to_owned()))
+            Ok(Some(TomlPackageBuild::SingleScript(BUILD_RS.to_owned())))
+        }
+        Some(TomlPackageBuild::MultipleScript(_scripts)) => {
+            if multiple_build_scripts {
+                unimplemented!("Multiple Build Scripts feature is not implemented yet!");
```

Comment:
Let's go ahead and do this in this commit

Reply:
Should be done